extract.py

```diff
@@ -120,6 +143,87 @@ def extract_database(args):
     return output_rows
 
 
+def extract_csv(args, mapping=None, init_id=0, date_format="%y%m%d", reg_rules=None):
+    if mapping is None:
+        mapping = dict()
+    if reg_rules is None:
+        reg_rules = dict()
+    translation = {"meta": {"reg_rules": reg_rules}, "mapping": mapping}
+    translation["meta"]["date_format"] = date_format
```

Comment:
Is there a reason for nesting `reg_rules` and `date_format` under the `meta` key?

The logic around ingesting the config seems a little odd, but maybe I'm missing something.

`main` loads the config and unpacks the entries into arguments with different names because `extract_csv` takes them as individual arguemnts

```python
def main():
    ....
       conf = json.load(f)
        ingest_kwargs = {
            "mapping": conf["translation_map"],
            "init_id": int(conf["initial_id"]),
            "date_format": conf["date_format"],
            "reg_rules": conf["regularization_rules"],
        }
       output_rows = extract_csv(args, **ingest_kwargs)
```

Then this method takes those arguments and repacks them into a new dict with it's own keys/nesting structure because translate rows requires them repacked into a dict

```python
def extract_csv(args, mapping=None, init_id=0, date_format="%y%m%d", reg_rules=None):
    ...
    translation = {"meta": {"reg_rules": reg_rules}, "mapping": mapping}
    translation["meta"]["date_format"] = date_format
    ...
    handled_row = translate_row(row, report, translation)
```

which it then immediately unpacks.

```python
def translate_row(row, report, translation):
    ...
    reg_rules = translation["meta"]["reg_rules"]
    mapping = translation["mapping"]
```

I'd either:
- pass the config through and use it directly in both `extact_csv` and `translate_rows`, without unpacking and repacking it in `main` and `extract_csv`
- unpack the config in main, and pass only the required fields down as arguments.